src/lib.rs

```diff
@@ -264,8 +267,26 @@ impl LightningNode {
         })
     }
 
-    pub fn get_node_info(&self) -> NodeInfo {
-        todo!()
+    pub fn get_node_info(&self) -> Result<NodeInfo> {
+        let node_state: NodeState = self.sdk.node_info().map_err(|e| RuntimeError {
+            code: RuntimeErrorCode::NodeUnavailable,
+            msg: e.to_string(),
+        })?;
+        let rate = self.get_exchange_rate();
+
+        Ok(NodeInfo {
+            node_pubkey: node_state.id,
+            peers: node_state.connected_peers,
+            channels_info: ChannelsInfo {
+                num_channels: 0,
+                num_usable_channels: 0,
+                local_balance: (node_state.channels_balance_msat + node_state.onchain_balance_msat)
+                    .to_amount_down(&rate),
```

Comment:
I wouldn't put the sum of these 2 balances in here. The name `local_balance` could indeed refer to both balances, but it breaks our existing model/assumptions. We have to think about how to manage the onchain balance, but for now, I would just do the following:
```suggestion
                local_balance: node_state.channels_balance_msat.to_amount_down(&rate),
```
We could already expose the onchain balance as a separate field, leaving to the consumer the decision to display the sum or not.